cabal-install/Distribution/Client/InstallPlan.hs

```diff
@@ -223,17 +223,21 @@ type InstallPlan = GenericInstallPlan
 
 -- | Smart constructor that deals with caching the 'Graph' representation.
 --
-mkInstallPlan :: Graph (GenericPlanPackage ipkg srcpkg)
+mkInstallPlan :: (IsUnit ipkg, IsUnit srcpkg)
```

Comment:
We can get more accurate location tracing by requiring a CallStack (WithCallStack)